transport/server.cc

```diff
@@ -378,23 +379,25 @@ cql_server::connection::read_frame() {
             if (_version < 3 || _version > current_version) {
                 auto client_version = _version;
                 _version = current_version;
-                throw exceptions::protocol_exception(format("Invalid or unsupported protocol version: {:d}", client_version));
+                return make_exception_future<ret_type>(exceptions::protocol_exception(format("Invalid or unsupported protocol version: {:d}", client_version)));
             }
 
-
             return _read_buf.read_exactly(frame_size() - 1).then([this] (temporary_buffer<char> tail) {
                 temporary_buffer<char> full(frame_size());
                 full.get_write()[0] = _version;
                 std::copy(tail.get(), tail.get() + tail.size(), full.get_write() + 1);
-                auto frame = parse_frame(std::move(full));
+                auto frame_res = parse_frame(std::move(full));
```

Comment:
nit: not sure if it's worth to rename frame variable. especially reading this code _res suffix means nothing to me.